### ISSUE TITLE
What should this be?

### ISSUE THREAD

**Issue by tbreloff** [JuliaPlot/juliaplot_docs#1]
Lets move the discussion from Gadfly.jl over to here.  What types of documents would be helpful for users??

**Comment by mbauman** [comment]
Thanks for doing this, Tom.  I think this can be a meaningful place for discussion and perhaps create a place for folks to donate abandoned plotting packages to the community.

That said, I worry that it will make any package that doesn't transfer here look second-class to a naive onlooker.

**Comment by tbreloff** [comment]
To quote myself: "I don't think anyone should move ownership of existing plotting packages here (yet)... right now lets just consolidate wikis/tutorials, and maybe we can link to it from the julia documentation."

I agree that there are lots of issues if we start transferring ownership... lets hash them out before we make any structural changes.

**Comment by waldyrious** [comment]
@mbauman I don't think that's the case, e.g. see:

- [JuliaGraphs](https://github.com/JuliaGraphs) vs. [GraphViz.jl](https://github.com/Keno/GraphViz.jl), [TikzGraphs.jl](https://github.com/sisl/TikzGraphs.jl) and [GraphLayout.jl](https://github.com/IainNZ/GraphLayout.jl)
- [JuliaWeb](https://github.com/JuliaWeb) vs. [Escher.jl](https://github.com/shashi/Escher.jl/issues/34)

If anything, it would be nice to hold off migration of packages until at least some consensus is found among the maintainers of the main plotting packages (I assume at least Gadfly.jl, PyPlot.jl, Winston.jl).

**Comment by Evizero** [comment]
> That said, I worry that it will make any package that doesn't transfer here look second-class to a naive onlooker.

Well, how this organization forks the big maintained packages like gadfly. This way all packages show up in a central repository

**Comment by Sisyphuss** [comment]
The first thing, IMHO, should be the "state of art" of Julia plot packages. It should give a brief or not so brief introduction to various packages and their state of development.

**Comment by mbauman** [comment]
:+1:.  I'd be very happy to be wrong.  I only brought it up because I definitely don't want this to step on any toes or make it feel like this is forcing someone's hand.  Limiting ownership transfer for now is a good idea.

I think the documentation here should be broad and thin.  Any "deeper" documentation should be submitted as patches to the individual packages, if they're needed.  One very interesting option would be an interactive PlotTutorial.jl package that walked folks through common usages of the major packages.

(I'm just on the periphery here — my little "plotting package" that brought me into the discussion is laughably minimal.)

**Comment by tbreloff** [comment]
Matt, I think your thoughts are exactly right.  Full documentation for
individual packages should live within the package itself.  JuliaPlot
should host summary documentation and tutorials that apply to the full
plotting landscape, with links to the individual packages for further
details.

As for your PlotTutorial.jl package idea, that's exactly the sort of thing
that I think belongs here.  People can submit examples/tutorials specific
to one package, and we as a community could help port the examples to other
packages for comparison.

**Comment by Evizero** [comment]
I do think though that something like julia-plot, that hasn't been touched in 4 years, should probably be removed from the list

**Comment by tbreloff** [comment]
I don't agree it should be removed, but we should briefly summarize the
package status for each (i.e. "No activity since 2012")... maybe put
abandoned packages in their own category.

**Comment by waldyrious** [comment]
I'm with  @tbreloff on this one. I was actually thinking of turning the list into a comparison table similar to those used in Wikipedia, where we could list what kinds of plots each package supports, whether they're METADATA-registered, when was the last release, etc.

**Comment by waldyrious** [comment]
> I definitely don't want this to step on any toes or make it feel like this is forcing someone's hand.

Exactly. I feel that if the main packages are eventually moved, there wouldn't be much resistance from smaller ones (although I may be wrong), but if we can't get those onboard, by moving only a few we'd indeed risk creating an unnecessary rift in the plotting packages landscape.

**Comment by Sisyphuss** [comment]
It's not important whether the authors move their packages here or not. One of the points of this organization is to give more visibility to those not so famous packages, which will benefit the Julia ecosystem in the long run.

**Comment by tbreloff** [comment]
I started a repo which I hope becomes the "plotting interface" that we've discussed:

https://github.com/tbreloff/Plot.jl

The idea is that it's too much to ask package maintainers to conform to a single interface, but that usually one interface can be converted to another by direct mapping.  My plan is to eventually map the core/common functionality for basic plotting from the Plot.jl interface to each package's style.  I waited to post until I had a bare minimum proof of concept... please check it out and let me know what you think, and also please post your interface wish-list at https://github.com/tbreloff/Plot.jl/issues/1.

Ideally this project will also help everyone understand where plotting functionality overlaps heavily between packages, where packages excel, and give us easy ways to do benchmark comparisons. 

It also will be great for those that like to do a lot of interactive plotting with one package, and then switch out the backend for a more publication-quality package when needed.

ps - I named it Plot.jl because, in my mind, this could be the "built-in" plotting that many new users want.  If you want it to be named JustAnotherCrappyPlotPackage.jl, please let me know.  I'm also open to moving it to JuliaPlot, but we just agreed not to move anything there yet.

**Comment by Evizero** [comment]
Love this ! Thanks for taking the first steps @tbreloff . This is in line with what I currently want to have in a non-plotting related package of mine, so I think I can spare time to code on this. 

The question now becomes what interface to adopt. I do think it will be difficult here and there to provide a common interface for PyPlot and Gadfly as they are very different.
However, since I am not a power-user of plotting utilities in Julia, I don't think I am that qualified to make interface suggestions. That being said, I do like the one-method-to-rule-them-all approach with `plot`. It should be simple and intuitive.

**Comment by Evizero** [comment]
Also, I think this package would be a perfect fit for JuliaPlot. So I would vote to move it

**Comment by waldyrious** [comment]
Glad you're working on the code side :) I think I might help most in the documentation side, e.g. by collecting a reasonably comprehensive set of common plotting examples and show how each package does it (in terms of syntax, output, etc.) in a single document. I believe this might help identifying the desirable syntax and direction for Plot.jl (which I find a totally appropriate name and agree with @Evizero would belong here even without moving any of the specific plotting packages).

I hope to be able to contribute that documentation soon.

**Comment by Evizero** [comment]
> by collecting a reasonably comprehensive set of common plotting examples and show how each package does it (in terms of syntax, output, etc.) in a single document. I believe this might help identifying the desirable syntax and direction for Plot.jl

That is a great idea!

**Comment by tbreloff** [comment]
@waldyrious: Compiling those examples would go a long way to helping me make progress with Plot.jl (since that's effectively what I'll have to do to implement a package). Looking forward to your docs!!

**Comment by Sisyphuss** [comment]
Concerning the interface, although one single uniform interface will be nice, it may be difficult or not realistic. In the latter case, we can provide several interfaces (styles) to ease the transmigration of users from other communities (Matlab, R, Python...) And maybe these high-level interfaces could be built on an intermediate-level interface beyond various packages (implementations).

**Comment by Evizero** [comment]
You do have a point, but we have to think realisticly in terms of manpower needed to implement and maintain the package. We have to get the ball rolling and get some momentum or we are likely to fail.

I think that the first milestone should be a very simple and intuitive porcelain interface for all the most common plot-types. That would be an important quick-win that would yield a high and visible benefit to new users.

**Comment by Sisyphuss** [comment]
What I said is not contradictory to yours. Let the middle-level interface be rich enough to embed most functionalities of most packages, and the high-level interface be concise in syntax and easy to use. What you mentioned is that currently only one high-level interface is enough. I agree with that.

**Comment by lobingera** [comment]
@tbreloff ,https://github.com/JuliaPlot/juliaplot_docs/issues/1#issuecomment-134597845

Did you think about moving the repository to here (JuliaPlot)?

**Comment by waldyrious** [comment]
@lobingera please read above:

>  I'm also open to moving it to JuliaPlot, but we just agreed not to move anything there yet.
--@tbreloff 
 
> I think this package would be a perfect fit for JuliaPlot. So I would vote to move it
--@Evizero

> I (...) agree with @Evizero would belong here even without moving any of the specific plotting packages
-- @waldyrious

**Comment by lobingera** [comment]
... i'm too slow ...

**Comment by waldyrious** [comment]
It's ok, but since this has happened before (e.g. in dcjones/Gadfly.jl#658), I'd kindly ask you to make sure to read threads carefully before commenting, otherwise you might repeat points already made / addressed, which adds noise and makes threads unnecessarily longer. Just a friendly reminder :)

**Comment by tbreloff** [comment]
Just a quick note... to try to keep with julia style, and avoid name clashing with Plot objects, I renamed my package: https://github.com/tbreloff/Plots.jl
